controllers/node-label/controller.go

```diff
@@ -0,0 +1,78 @@
+package nodelabel
+
+import (
+	"context"
+	"reflect"
+
+	msyncv1 "github.com/darkowlzz/operator-toolkit/controller/metadata-sync/v1"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Controller implements the Sync contoller interface, applying node labels to
+// StorageOS nodes.
+type Controller struct {
+	api NodeLabeller
+	log logr.Logger
+}
+
+var _ msyncv1.Controller = &Controller{}
+
+// NewController returns a Controller that implements node label sync in
+// StorageOS.
+func NewController(api NodeLabeller, log logr.Logger) (*Controller, error) {
+	return &Controller{api: api, log: log}, nil
+}
+
+// Ensure applies labels set on the k8s node to the StorageOS node.
+//
+// StorageOS reserved labels are validated and applied first, then the remaining
+// unreserved labels are applied.
+//
+// Any errors will result in a requeue, with standard back-off retries.
+//
+// There is no label sync from StorageOS to Kubernetes.  This is intentional to
+// ensure a simple flow of desired state set by users in Kubernetes to actual
+// state set on the StorageOS node.
+func (c Controller) Ensure(ctx context.Context, obj client.Object) error {
+	if err := c.api.EnsureNodeLabels(obj.GetName(), obj.GetLabels()); err != nil {
```

Comment:
Can we update the `EnsureNodeLabels()` function signature to accept the context and pass it forward?
I see `EnsureUnreservedNodeLabels()` and `EnsureComputeOnly()` creating their own context.

Reply:
As discussed, I've done just for the functions used in this PR.  I'll open a separate PR to apply the function signature consistently across the rest.